proposal/ran-ablaze/README.rst

```diff
@@ -0,0 +1,94 @@
+===================================================================
+Simple Polyhedral Structure and Optimization
+===================================================================
+
+The GitHub repository: 
+
+Basic Information
+=================
+
+The polyhedral model is a powerful mathematical framework that reasons about loops in programs. Each lattice point inside a polyhedral corresponds with one instance of statement within nested loops.
+
+Problem to Solve
+================
+
+The interest of using polyhedral representations is that they can be manipulated or optimized with algorithms whose complexity depends on their structure. Furthermore, generic and compact solutions can be designed that depend on program parameters. 
+
+The Simple Polyhedral Structure tries to use C++ to construct a simple API to express the model and how the model should be optimized.
+
+The structure has a scheduling language featuring novel commands for targeting multiple high performance architectures. It is designed to enable easy integration of code generators for new architectures.
+
+Prospective Users
+=================
+
+Almost all loops in a majority of High Performance Computing programs and deep learning.
+
+System Architecture
+===================
+
+Simple Polyhedral Structure can use simple instructions to declare loops and generate specified optimized programs.
+
+The `iter` class represents the iterator variables. The range defines the range of the loop around the computation.
+
+The `expr` class represents the expressions.
+
+The `comput` class represents the computations that share schedules up to a level. A computation has an expression and iteration domain defined using an iterator variable. It contains many optimization functions.
+
+Workflow:
+
+1. Declare iterator variables
+2. Declare expressions
+3. Specify optimizations
+4. Generate code
```

Comment:
Users of computation-intensive code care a lot about performance.  They are willing to write assembly as long as it runs faster.  You need to convince them that your tool makes the loops faster or they will not use it.

If you only demonstrate the automatic optimization works, it becomes more like a compiler project rather than a computing one.

Reply:
Although I call it an optimized function, it actually uses a polyhedral model to do a series of affine transformation, which I think is a computing project